xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -251,31 +253,19 @@ func (b *clusterImplBalancer) UpdateClientConnState(s balancer.ClientConnState)
 		return err
 	}
 
-	// If child policy is a different type, recreate the sub-balancer.
 	if b.config == nil || b.config.ChildPolicy.Name != newConfig.ChildPolicy.Name {
-		if b.childLB != nil {
-			b.childLB.Close()
+		if err := b.child.SwitchTo(bb); err != nil {
+			return fmt.Errorf("cluster impl: error switching to child of type %q: %v", newConfig.ChildPolicy.Name, err)
```

Comment:
Nit: The existing error strings don't have a prefix of `cluster impl`. My preference would be to stay consistent with existing code or switch them all. And even if you switch, I would think the prefix would be `clusterimpl` and not `cluster impl`.

Reply:
Dropped the prefix.